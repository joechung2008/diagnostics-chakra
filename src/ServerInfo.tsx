import { Table } from "@chakra-ui/react";

const ServerInfo: React.FC<ServerInfoProps> = ({
  deploymentId,
  extensionSync,
  hostname,
  nodeVersions,
  serverId,
  uptime,
}) => {
  const items = [
    {
      name: "Hostname",
      value: hostname,
    },
    {
      name: "Uptime",
      value: uptime,
    },
    {
      name: "Server ID",
      value: serverId,
    },
    {
      name: "Deployment ID",
      value: deploymentId,
    },
    {
      name: "Node Versions",
      value: nodeVersions,
    },
    {
      name: "Extension Sync | Total Sync All Count",
      value: extensionSync.totalSyncAllCount,
    },
  ];

  return (
    <Table.Root>
      <Table.Header>
        <Table.Row>
          <Table.ColumnHeader>Name</Table.ColumnHeader>
          <Table.ColumnHeader>Value</Table.ColumnHeader>
        </Table.Row>
      </Table.Header>
      <Table.Body>
        {items.map((item) => (
          <Table.Row key={item.name}>
            <Table.Cell>{item.name}</Table.Cell>
            <Table.Cell>{item.value}</Table.Cell>
          </Table.Row>
        ))}
      </Table.Body>
    </Table.Root>
  );
};

export default ServerInfo;
