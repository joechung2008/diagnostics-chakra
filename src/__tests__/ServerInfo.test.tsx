import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import ServerInfo from "../ServerInfo";
import { TestWrapper } from "./test-utils";

describe("ServerInfo", () => {
  const mockServerInfo = {
    deploymentId: "deploy-123",
    extensionSync: {
      totalSyncAllCount: 42,
    },
    hostname: "server.example.com",
    nodeVersions: "v18.17.0",
    serverId: "server-456",
    uptime: 1234567890,
  };

  it("should render all server information fields", () => {
    const { asFragment } = render(
      <TestWrapper>
        <ServerInfo {...mockServerInfo} />
      </TestWrapper>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("should render table with correct headers", () => {
    const { asFragment } = render(
      <TestWrapper>
        <ServerInfo {...mockServerInfo} />
      </TestWrapper>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("should render correct number of rows", () => {
    const { asFragment } = render(
      <TestWrapper>
        <ServerInfo {...mockServerInfo} />
      </TestWrapper>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("should handle different data types", () => {
    const testData = {
      ...mockServerInfo,
      uptime: 987654321,
      nodeVersions: "v20.10.0",
      extensionSync: {
        totalSyncAllCount: 0,
      },
    };

    const { asFragment } = render(
      <TestWrapper>
        <ServerInfo {...testData} />
      </TestWrapper>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("should handle long hostnames", () => {
    const longHostname = "very-long-hostname-server.example.com";
    const testData = {
      ...mockServerInfo,
      hostname: longHostname,
    };

    const { asFragment } = render(
      <TestWrapper>
        <ServerInfo {...testData} />
      </TestWrapper>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("should handle special characters in values", () => {
    const testData = {
      ...mockServerInfo,
      hostname: "server_with_underscores.example.com",
      nodeVersions: "v18.17.0 (with extra info)",
    };

    const { asFragment } = render(
      <TestWrapper>
        <ServerInfo {...testData} />
      </TestWrapper>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
