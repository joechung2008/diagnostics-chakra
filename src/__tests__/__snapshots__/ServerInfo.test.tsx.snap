// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ServerInfo > should handle different data types 1`] = `
<DocumentFragment>
  <script
    nonce=""
  >
    ((e,i,s,u,m,a,l,h)=&gt;{let d=document.documentElement,w=["light","dark"];function p(n){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&a?m.map(f=&gt;a[f]||f):m;k?(d.classList.remove(...S),d.classList.add(a&&a[n]?a[n]:n)):d.setAttribute(y,n)}),R(n)}function R(n){h&&w.includes(n)&&(d.style.colorScheme=n)}function c(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(u)p(u);else try{let n=localStorage.getItem(i)||s,y=l&&n==="system"?c():n;p(y)}catch(n){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <table
    class="chakra-table__root css-a2tetu"
  >
    <thead
      class="chakra-table__header"
    >
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <th
          class="chakra-table__columnHeader css-kim7sg"
        >
          Name
        </th>
        <th
          class="chakra-table__columnHeader css-kim7sg"
        >
          Value
        </th>
      </tr>
    </thead>
    <tbody
      class="chakra-table__body"
    >
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <td
          class="chakra-table__cell css-1luypx0"
        >
          Hostname
        </td>
        <td
          class="chakra-table__cell css-1luypx0"
        >
          server.example.com
        </td>
      </tr>
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <td
          class="chakra-table__cell css-1luypx0"
        >
          Uptime
        </td>
        <td
          class="chakra-table__cell css-1luypx0"
        >
          987654321
        </td>
      </tr>
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <td
          class="chakra-table__cell css-1luypx0"
        >
          Server ID
        </td>
        <td
          class="chakra-table__cell css-1luypx0"
        >
          server-456
        </td>
      </tr>
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <td
          class="chakra-table__cell css-1luypx0"
        >
          Deployment ID
        </td>
        <td
          class="chakra-table__cell css-1luypx0"
        >
          deploy-123
        </td>
      </tr>
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <td
          class="chakra-table__cell css-1luypx0"
        >
          Node Versions
        </td>
        <td
          class="chakra-table__cell css-1luypx0"
        >
          v20.10.0
        </td>
      </tr>
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <td
          class="chakra-table__cell css-1luypx0"
        >
          Extension Sync | Total Sync All Count
        </td>
        <td
          class="chakra-table__cell css-1luypx0"
        >
          0
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`ServerInfo > should handle long hostnames 1`] = `
<DocumentFragment>
  <script
    nonce=""
  >
    ((e,i,s,u,m,a,l,h)=&gt;{let d=document.documentElement,w=["light","dark"];function p(n){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&a?m.map(f=&gt;a[f]||f):m;k?(d.classList.remove(...S),d.classList.add(a&&a[n]?a[n]:n)):d.setAttribute(y,n)}),R(n)}function R(n){h&&w.includes(n)&&(d.style.colorScheme=n)}function c(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(u)p(u);else try{let n=localStorage.getItem(i)||s,y=l&&n==="system"?c():n;p(y)}catch(n){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <table
    class="chakra-table__root css-a2tetu"
  >
    <thead
      class="chakra-table__header"
    >
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <th
          class="chakra-table__columnHeader css-kim7sg"
        >
          Name
        </th>
        <th
          class="chakra-table__columnHeader css-kim7sg"
        >
          Value
        </th>
      </tr>
    </thead>
    <tbody
      class="chakra-table__body"
    >
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <td
          class="chakra-table__cell css-1luypx0"
        >
          Hostname
        </td>
        <td
          class="chakra-table__cell css-1luypx0"
        >
          very-long-hostname-server.example.com
        </td>
      </tr>
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <td
          class="chakra-table__cell css-1luypx0"
        >
          Uptime
        </td>
        <td
          class="chakra-table__cell css-1luypx0"
        >
          1234567890
        </td>
      </tr>
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <td
          class="chakra-table__cell css-1luypx0"
        >
          Server ID
        </td>
        <td
          class="chakra-table__cell css-1luypx0"
        >
          server-456
        </td>
      </tr>
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <td
          class="chakra-table__cell css-1luypx0"
        >
          Deployment ID
        </td>
        <td
          class="chakra-table__cell css-1luypx0"
        >
          deploy-123
        </td>
      </tr>
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <td
          class="chakra-table__cell css-1luypx0"
        >
          Node Versions
        </td>
        <td
          class="chakra-table__cell css-1luypx0"
        >
          v18.17.0
        </td>
      </tr>
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <td
          class="chakra-table__cell css-1luypx0"
        >
          Extension Sync | Total Sync All Count
        </td>
        <td
          class="chakra-table__cell css-1luypx0"
        >
          42
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`ServerInfo > should handle special characters in values 1`] = `
<DocumentFragment>
  <script
    nonce=""
  >
    ((e,i,s,u,m,a,l,h)=&gt;{let d=document.documentElement,w=["light","dark"];function p(n){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&a?m.map(f=&gt;a[f]||f):m;k?(d.classList.remove(...S),d.classList.add(a&&a[n]?a[n]:n)):d.setAttribute(y,n)}),R(n)}function R(n){h&&w.includes(n)&&(d.style.colorScheme=n)}function c(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(u)p(u);else try{let n=localStorage.getItem(i)||s,y=l&&n==="system"?c():n;p(y)}catch(n){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <table
    class="chakra-table__root css-a2tetu"
  >
    <thead
      class="chakra-table__header"
    >
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <th
          class="chakra-table__columnHeader css-kim7sg"
        >
          Name
        </th>
        <th
          class="chakra-table__columnHeader css-kim7sg"
        >
          Value
        </th>
      </tr>
    </thead>
    <tbody
      class="chakra-table__body"
    >
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <td
          class="chakra-table__cell css-1luypx0"
        >
          Hostname
        </td>
        <td
          class="chakra-table__cell css-1luypx0"
        >
          server_with_underscores.example.com
        </td>
      </tr>
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <td
          class="chakra-table__cell css-1luypx0"
        >
          Uptime
        </td>
        <td
          class="chakra-table__cell css-1luypx0"
        >
          1234567890
        </td>
      </tr>
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <td
          class="chakra-table__cell css-1luypx0"
        >
          Server ID
        </td>
        <td
          class="chakra-table__cell css-1luypx0"
        >
          server-456
        </td>
      </tr>
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <td
          class="chakra-table__cell css-1luypx0"
        >
          Deployment ID
        </td>
        <td
          class="chakra-table__cell css-1luypx0"
        >
          deploy-123
        </td>
      </tr>
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <td
          class="chakra-table__cell css-1luypx0"
        >
          Node Versions
        </td>
        <td
          class="chakra-table__cell css-1luypx0"
        >
          v18.17.0 (with extra info)
        </td>
      </tr>
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <td
          class="chakra-table__cell css-1luypx0"
        >
          Extension Sync | Total Sync All Count
        </td>
        <td
          class="chakra-table__cell css-1luypx0"
        >
          42
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`ServerInfo > should render all server information fields 1`] = `
<DocumentFragment>
  <script
    nonce=""
  >
    ((e,i,s,u,m,a,l,h)=&gt;{let d=document.documentElement,w=["light","dark"];function p(n){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&a?m.map(f=&gt;a[f]||f):m;k?(d.classList.remove(...S),d.classList.add(a&&a[n]?a[n]:n)):d.setAttribute(y,n)}),R(n)}function R(n){h&&w.includes(n)&&(d.style.colorScheme=n)}function c(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(u)p(u);else try{let n=localStorage.getItem(i)||s,y=l&&n==="system"?c():n;p(y)}catch(n){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <table
    class="chakra-table__root css-a2tetu"
  >
    <thead
      class="chakra-table__header"
    >
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <th
          class="chakra-table__columnHeader css-kim7sg"
        >
          Name
        </th>
        <th
          class="chakra-table__columnHeader css-kim7sg"
        >
          Value
        </th>
      </tr>
    </thead>
    <tbody
      class="chakra-table__body"
    >
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <td
          class="chakra-table__cell css-1luypx0"
        >
          Hostname
        </td>
        <td
          class="chakra-table__cell css-1luypx0"
        >
          server.example.com
        </td>
      </tr>
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <td
          class="chakra-table__cell css-1luypx0"
        >
          Uptime
        </td>
        <td
          class="chakra-table__cell css-1luypx0"
        >
          1234567890
        </td>
      </tr>
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <td
          class="chakra-table__cell css-1luypx0"
        >
          Server ID
        </td>
        <td
          class="chakra-table__cell css-1luypx0"
        >
          server-456
        </td>
      </tr>
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <td
          class="chakra-table__cell css-1luypx0"
        >
          Deployment ID
        </td>
        <td
          class="chakra-table__cell css-1luypx0"
        >
          deploy-123
        </td>
      </tr>
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <td
          class="chakra-table__cell css-1luypx0"
        >
          Node Versions
        </td>
        <td
          class="chakra-table__cell css-1luypx0"
        >
          v18.17.0
        </td>
      </tr>
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <td
          class="chakra-table__cell css-1luypx0"
        >
          Extension Sync | Total Sync All Count
        </td>
        <td
          class="chakra-table__cell css-1luypx0"
        >
          42
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`ServerInfo > should render correct number of rows 1`] = `
<DocumentFragment>
  <script
    nonce=""
  >
    ((e,i,s,u,m,a,l,h)=&gt;{let d=document.documentElement,w=["light","dark"];function p(n){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&a?m.map(f=&gt;a[f]||f):m;k?(d.classList.remove(...S),d.classList.add(a&&a[n]?a[n]:n)):d.setAttribute(y,n)}),R(n)}function R(n){h&&w.includes(n)&&(d.style.colorScheme=n)}function c(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(u)p(u);else try{let n=localStorage.getItem(i)||s,y=l&&n==="system"?c():n;p(y)}catch(n){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <table
    class="chakra-table__root css-a2tetu"
  >
    <thead
      class="chakra-table__header"
    >
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <th
          class="chakra-table__columnHeader css-kim7sg"
        >
          Name
        </th>
        <th
          class="chakra-table__columnHeader css-kim7sg"
        >
          Value
        </th>
      </tr>
    </thead>
    <tbody
      class="chakra-table__body"
    >
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <td
          class="chakra-table__cell css-1luypx0"
        >
          Hostname
        </td>
        <td
          class="chakra-table__cell css-1luypx0"
        >
          server.example.com
        </td>
      </tr>
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <td
          class="chakra-table__cell css-1luypx0"
        >
          Uptime
        </td>
        <td
          class="chakra-table__cell css-1luypx0"
        >
          1234567890
        </td>
      </tr>
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <td
          class="chakra-table__cell css-1luypx0"
        >
          Server ID
        </td>
        <td
          class="chakra-table__cell css-1luypx0"
        >
          server-456
        </td>
      </tr>
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <td
          class="chakra-table__cell css-1luypx0"
        >
          Deployment ID
        </td>
        <td
          class="chakra-table__cell css-1luypx0"
        >
          deploy-123
        </td>
      </tr>
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <td
          class="chakra-table__cell css-1luypx0"
        >
          Node Versions
        </td>
        <td
          class="chakra-table__cell css-1luypx0"
        >
          v18.17.0
        </td>
      </tr>
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <td
          class="chakra-table__cell css-1luypx0"
        >
          Extension Sync | Total Sync All Count
        </td>
        <td
          class="chakra-table__cell css-1luypx0"
        >
          42
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`ServerInfo > should render table with correct headers 1`] = `
<DocumentFragment>
  <script
    nonce=""
  >
    ((e,i,s,u,m,a,l,h)=&gt;{let d=document.documentElement,w=["light","dark"];function p(n){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&a?m.map(f=&gt;a[f]||f):m;k?(d.classList.remove(...S),d.classList.add(a&&a[n]?a[n]:n)):d.setAttribute(y,n)}),R(n)}function R(n){h&&w.includes(n)&&(d.style.colorScheme=n)}function c(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(u)p(u);else try{let n=localStorage.getItem(i)||s,y=l&&n==="system"?c():n;p(y)}catch(n){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <table
    class="chakra-table__root css-a2tetu"
  >
    <thead
      class="chakra-table__header"
    >
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <th
          class="chakra-table__columnHeader css-kim7sg"
        >
          Name
        </th>
        <th
          class="chakra-table__columnHeader css-kim7sg"
        >
          Value
        </th>
      </tr>
    </thead>
    <tbody
      class="chakra-table__body"
    >
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <td
          class="chakra-table__cell css-1luypx0"
        >
          Hostname
        </td>
        <td
          class="chakra-table__cell css-1luypx0"
        >
          server.example.com
        </td>
      </tr>
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <td
          class="chakra-table__cell css-1luypx0"
        >
          Uptime
        </td>
        <td
          class="chakra-table__cell css-1luypx0"
        >
          1234567890
        </td>
      </tr>
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <td
          class="chakra-table__cell css-1luypx0"
        >
          Server ID
        </td>
        <td
          class="chakra-table__cell css-1luypx0"
        >
          server-456
        </td>
      </tr>
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <td
          class="chakra-table__cell css-1luypx0"
        >
          Deployment ID
        </td>
        <td
          class="chakra-table__cell css-1luypx0"
        >
          deploy-123
        </td>
      </tr>
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <td
          class="chakra-table__cell css-1luypx0"
        >
          Node Versions
        </td>
        <td
          class="chakra-table__cell css-1luypx0"
        >
          v18.17.0
        </td>
      </tr>
      <tr
        class="chakra-table__row css-11ot24d"
      >
        <td
          class="chakra-table__cell css-1luypx0"
        >
          Extension Sync | Total Sync All Count
        </td>
        <td
          class="chakra-table__cell css-1luypx0"
        >
          42
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
